tests/testsuite/metadata.rs

```diff
@@ -4954,3 +4954,34 @@ local-time = 1979-05-27
         )
         .run();
 }
+
+#[cargo_test]
+fn metadata_ignores_build_target_configuration() -> anyhow::Result<()> {
```

Comment:
nit: we prefer contributions split tests into the first commit, with them passing.  That way the fix commit will show the change in behavior by how the test changes.  This helps test the test and provides an illustration of the PR description for the reviewer and the community at large.

We linked to examples in the [contrib docs explaining this](https://doc.crates.io/contrib/process/working-on-cargo.html#submitting-a-pull-request)